src/hooks/usePositions.tsx

```diff
@@ -44,9 +60,47 @@ export const usePositions = (searchPositionId: string) => {
 
   React.useEffect(() => {
     if (positionsData) {
-      setData(marshalPositionListData(positionsData.positions, userData?.user))
+      const positionListData = marshalPositionListData(positionsData.positions, userData?.user)
+
+      const fetchUserBalanceWithDecimals = async () => {
```

Comment:
I think this can be improved. 
Lets suppose that every collateral token from positionListData is not listed in configs and is the same for every position, like
```typescript
positionListData = [
  {
     id: 1
     ...,
     collateralToken: '0x123'
  },
{
     id: 2
     ...,
     collateralToken: '0x123'
  },
{
     id: 3
     ...,
     collateralToken: '0x123'
  },
  ...
]
```
We'll call `ERC20Service@0x123 getProfileSummary` for each element in the list, when we only need to call it once.

One solution could be to extract unique collateral tokens from positions list and then call `getProfileSummary` for each of them and with the resulting data create a lookup object by their ids that we can use later for calculating `userBalanceWithDecimals` (maybe inside `marshalPositionListData`)

Reply:
I fixed this with a  map and using uniqby package. I think is like an enhanced, so I keep the _marshalPositionListData_ with his previous responsability